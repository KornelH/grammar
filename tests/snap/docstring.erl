-module(docstring).

-export([f/0]).

-doc """
     Docstring examples
     """.
f() ->
    """
  Line "1"
  Line "2"
  """ = "Line \"1\"
Line \"2\"",
    X = lists:seq(1,3), % just to check is syntax highlight is still ok

    """
      First line starting with two spaces
    Not escaped: "\t \r \xFF" and """

    """ = "  First line starting with two spaces
Not escaped: \"\\t \\r \\xFF\" and \"\"\"
",
    X = lists:seq(1,3), % just to check is syntax highlight is still ok

    """""
""""
FIXME: previous line is not a closing delimiter because opening-closing delimiter have five double quote characters
""""" = "\"\"\"\"",
    X = lists:seq(1,3), % just to check is syntax highlight is still ok

    ok.

-define(THIS_IS_THE_END, "end"). % just to check is syntax highlight is still ok
