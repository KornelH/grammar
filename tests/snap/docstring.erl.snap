>-module(docstring).
#^ source.erlang meta.directive.module.erlang punctuation.section.directive.begin.erlang
# ^^^^^^ source.erlang meta.directive.module.erlang keyword.control.directive.module.erlang
#       ^ source.erlang meta.directive.module.erlang punctuation.definition.parameters.begin.erlang
#        ^^^^^^^^^ source.erlang meta.directive.module.erlang entity.name.type.class.module.definition.erlang
#                 ^ source.erlang meta.directive.module.erlang punctuation.definition.parameters.end.erlang
#                  ^ source.erlang meta.directive.module.erlang punctuation.section.directive.end.erlang
>
>-export([f/0]).
#^ source.erlang meta.directive.export.erlang punctuation.section.directive.begin.erlang
# ^^^^^^ source.erlang meta.directive.export.erlang keyword.control.directive.export.erlang
#       ^ source.erlang meta.directive.export.erlang punctuation.definition.parameters.begin.erlang
#        ^ source.erlang meta.directive.export.erlang meta.structure.list.function.erlang punctuation.definition.list.begin.erlang
#         ^ source.erlang meta.directive.export.erlang meta.structure.list.function.erlang entity.name.function.erlang
#          ^ source.erlang meta.directive.export.erlang meta.structure.list.function.erlang punctuation.separator.function-arity.erlang
#           ^ source.erlang meta.directive.export.erlang meta.structure.list.function.erlang constant.numeric.integer.decimal.erlang
#            ^ source.erlang meta.directive.export.erlang meta.structure.list.function.erlang punctuation.definition.list.end.erlang
#             ^ source.erlang meta.directive.export.erlang punctuation.definition.parameters.end.erlang
#              ^ source.erlang meta.directive.export.erlang punctuation.section.directive.end.erlang
>
>-doc """
#^ source.erlang meta.directive.erlang punctuation.section.directive.begin.erlang
# ^^^ source.erlang meta.directive.erlang keyword.control.directive.erlang
#    ^ source.erlang meta.directive.erlang
#     ^^^^ source.erlang meta.directive.erlang string.string.docstring.multi.erlang punctuation.definition.docstring.begin.erlang
>     Docstring examples
#^^^^^^^^^^^^^^^^^^^^^^^^ source.erlang meta.directive.erlang string.string.docstring.multi.erlang
>     """.
#^^^^^^^^ source.erlang meta.directive.erlang string.string.docstring.multi.erlang punctuation.definition.docstring.end.erlang
#        ^ source.erlang meta.directive.erlang punctuation.section.directive.end.erlang
>f() ->
#^ source.erlang meta.function.erlang entity.name.function.definition.erlang
# ^ source.erlang meta.function.erlang meta.expression.parenthesized punctuation.section.expression.begin.erlang
#  ^ source.erlang meta.function.erlang meta.expression.parenthesized punctuation.section.expression.end.erlang
#   ^ source.erlang meta.function.erlang
#    ^ source.erlang meta.function.erlang keyword.operator.symbolic.erlang
#     ^ source.erlang meta.function.erlang keyword.operator.symbolic.erlang
>    """
#^^^^ source.erlang meta.function.erlang
#    ^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang punctuation.definition.docstring.begin.erlang
>  Line "1"
#^^^^^^^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang
>  Line "2"
#^^^^^^^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang
>  """ = "Line \"1\"
#^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang punctuation.definition.docstring.end.erlang
#     ^ source.erlang meta.function.erlang
#      ^ source.erlang meta.function.erlang keyword.operator.symbolic.erlang
#       ^ source.erlang meta.function.erlang
#        ^ source.erlang meta.function.erlang string.quoted.double.erlang punctuation.definition.string.begin.erlang
#         ^^^^^ source.erlang meta.function.erlang string.quoted.double.erlang
#              ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#               ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#                ^ source.erlang meta.function.erlang string.quoted.double.erlang
#                 ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#                  ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
>Line \"2\"",
#^^^^^ source.erlang meta.function.erlang string.quoted.double.erlang
#     ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#      ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#       ^ source.erlang meta.function.erlang string.quoted.double.erlang
#        ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#         ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#          ^ source.erlang meta.function.erlang string.quoted.double.erlang punctuation.definition.string.end.erlang
#           ^ source.erlang meta.function.erlang punctuation.separator.expressions.erlang
>    X = lists:seq(1,3), % just to check is syntax highlight is still ok
#^^^^ source.erlang meta.function.erlang
#    ^ source.erlang meta.function.erlang variable.other.erlang
#     ^ source.erlang meta.function.erlang
#      ^ source.erlang meta.function.erlang keyword.operator.symbolic.erlang
#       ^ source.erlang meta.function.erlang
#        ^^^^^ source.erlang meta.function.erlang meta.function-call.erlang entity.name.type.class.module.erlang
#             ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.separator.module-function.erlang
#              ^^^ source.erlang meta.function.erlang meta.function-call.erlang entity.name.function.erlang
#                 ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.definition.parameters.begin.erlang
#                  ^ source.erlang meta.function.erlang meta.function-call.erlang constant.numeric.integer.decimal.erlang
#                   ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.separator.parameters.erlang
#                    ^ source.erlang meta.function.erlang meta.function-call.erlang constant.numeric.integer.decimal.erlang
#                     ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.definition.parameters.end.erlang
#                      ^ source.erlang meta.function.erlang punctuation.separator.expressions.erlang
#                       ^ source.erlang meta.function.erlang
#                        ^ source.erlang meta.function.erlang comment.line.percentage.erlang punctuation.definition.comment.erlang
#                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.erlang meta.function.erlang comment.line.percentage.erlang
>
>    """
#^^^^ source.erlang meta.function.erlang
#    ^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang punctuation.definition.docstring.begin.erlang
>      First line starting with two spaces
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang
>    Not escaped: "\t \r \xFF" and """
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang
>
>    """ = "  First line starting with two spaces
#^^^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang punctuation.definition.docstring.end.erlang
#       ^ source.erlang meta.function.erlang
#        ^ source.erlang meta.function.erlang keyword.operator.symbolic.erlang
#         ^ source.erlang meta.function.erlang
#          ^ source.erlang meta.function.erlang string.quoted.double.erlang punctuation.definition.string.begin.erlang
#           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.erlang meta.function.erlang string.quoted.double.erlang
>Not escaped: \"\\t \\r \\xFF\" and \"\"\"
#^^^^^^^^^^^^^ source.erlang meta.function.erlang string.quoted.double.erlang
#             ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#              ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#               ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#                ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#                 ^^ source.erlang meta.function.erlang string.quoted.double.erlang
#                   ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#                    ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#                     ^^ source.erlang meta.function.erlang string.quoted.double.erlang
#                       ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#                        ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#                         ^^^ source.erlang meta.function.erlang string.quoted.double.erlang
#                            ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#                             ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#                              ^^^^^ source.erlang meta.function.erlang string.quoted.double.erlang
#                                   ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#                                    ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#                                     ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#                                      ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#                                       ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#                                        ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
>",
#^ source.erlang meta.function.erlang string.quoted.double.erlang punctuation.definition.string.end.erlang
# ^ source.erlang meta.function.erlang punctuation.separator.expressions.erlang
>    X = lists:seq(1,3), % just to check is syntax highlight is still ok
#^^^^ source.erlang meta.function.erlang
#    ^ source.erlang meta.function.erlang variable.other.erlang
#     ^ source.erlang meta.function.erlang
#      ^ source.erlang meta.function.erlang keyword.operator.symbolic.erlang
#       ^ source.erlang meta.function.erlang
#        ^^^^^ source.erlang meta.function.erlang meta.function-call.erlang entity.name.type.class.module.erlang
#             ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.separator.module-function.erlang
#              ^^^ source.erlang meta.function.erlang meta.function-call.erlang entity.name.function.erlang
#                 ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.definition.parameters.begin.erlang
#                  ^ source.erlang meta.function.erlang meta.function-call.erlang constant.numeric.integer.decimal.erlang
#                   ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.separator.parameters.erlang
#                    ^ source.erlang meta.function.erlang meta.function-call.erlang constant.numeric.integer.decimal.erlang
#                     ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.definition.parameters.end.erlang
#                      ^ source.erlang meta.function.erlang punctuation.separator.expressions.erlang
#                       ^ source.erlang meta.function.erlang
#                        ^ source.erlang meta.function.erlang comment.line.percentage.erlang punctuation.definition.comment.erlang
#                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.erlang meta.function.erlang comment.line.percentage.erlang
>
>    """""
#^^^^ source.erlang meta.function.erlang
#    ^^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang punctuation.definition.docstring.begin.erlang
>""""
#^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang
>FIXME: previous line is not a closing delimiter because opening-closing delimiter have five double quote characters
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang
>""""" = "\"\"\"\"",
#^^^^^ source.erlang meta.function.erlang string.string.docstring.multi.erlang punctuation.definition.docstring.end.erlang
#     ^ source.erlang meta.function.erlang
#      ^ source.erlang meta.function.erlang keyword.operator.symbolic.erlang
#       ^ source.erlang meta.function.erlang
#        ^ source.erlang meta.function.erlang string.quoted.double.erlang punctuation.definition.string.begin.erlang
#         ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#          ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#           ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#            ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#             ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#              ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#               ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang punctuation.definition.escape.erlang
#                ^ source.erlang meta.function.erlang string.quoted.double.erlang constant.character.escape.erlang
#                 ^ source.erlang meta.function.erlang string.quoted.double.erlang punctuation.definition.string.end.erlang
#                  ^ source.erlang meta.function.erlang punctuation.separator.expressions.erlang
>    X = lists:seq(1,3), % just to check is syntax highlight is still ok
#^^^^ source.erlang meta.function.erlang
#    ^ source.erlang meta.function.erlang variable.other.erlang
#     ^ source.erlang meta.function.erlang
#      ^ source.erlang meta.function.erlang keyword.operator.symbolic.erlang
#       ^ source.erlang meta.function.erlang
#        ^^^^^ source.erlang meta.function.erlang meta.function-call.erlang entity.name.type.class.module.erlang
#             ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.separator.module-function.erlang
#              ^^^ source.erlang meta.function.erlang meta.function-call.erlang entity.name.function.erlang
#                 ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.definition.parameters.begin.erlang
#                  ^ source.erlang meta.function.erlang meta.function-call.erlang constant.numeric.integer.decimal.erlang
#                   ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.separator.parameters.erlang
#                    ^ source.erlang meta.function.erlang meta.function-call.erlang constant.numeric.integer.decimal.erlang
#                     ^ source.erlang meta.function.erlang meta.function-call.erlang punctuation.definition.parameters.end.erlang
#                      ^ source.erlang meta.function.erlang punctuation.separator.expressions.erlang
#                       ^ source.erlang meta.function.erlang
#                        ^ source.erlang meta.function.erlang comment.line.percentage.erlang punctuation.definition.comment.erlang
#                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.erlang meta.function.erlang comment.line.percentage.erlang
>
>    ok.
#^^^^ source.erlang meta.function.erlang
#    ^^ source.erlang meta.function.erlang constant.other.symbol.unquoted.erlang
#      ^ source.erlang meta.function.erlang punctuation.terminator.function.erlang
>
>-define(THIS_IS_THE_END, "end"). % just to check is syntax highlight is still ok
#^ source.erlang meta.directive.define.erlang punctuation.section.directive.begin.erlang
# ^^^^^^ source.erlang meta.directive.define.erlang keyword.control.directive.define.erlang
#       ^ source.erlang meta.directive.define.erlang punctuation.definition.parameters.begin.erlang
#        ^^^^^^^^^^^^^^^ source.erlang meta.directive.define.erlang entity.name.function.macro.definition.erlang
#                       ^^ source.erlang meta.directive.define.erlang
#                         ^ source.erlang meta.directive.define.erlang string.quoted.double.erlang punctuation.definition.string.begin.erlang
#                          ^^^ source.erlang meta.directive.define.erlang string.quoted.double.erlang
#                             ^ source.erlang meta.directive.define.erlang string.quoted.double.erlang punctuation.definition.string.end.erlang
#                              ^ source.erlang meta.directive.define.erlang punctuation.definition.parameters.end.erlang
#                               ^ source.erlang meta.directive.define.erlang punctuation.section.directive.end.erlang
#                                ^ source.erlang
#                                 ^ source.erlang comment.line.percentage.erlang punctuation.definition.comment.erlang
#                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.erlang comment.line.percentage.erlang
>